### PR TITLE
Write XML to JSON flow that roughtly follows JAXB+Jackson semantics

### PR DESCRIPTION
**Pull request by jypma**
@alar17 please review. We need this to support our existing JS-land validation code.

### CODE REVIEW

File: ts-reaktive-marshal-xerces/src/main/java/com/tradeshift/reaktive/marshal/convert/XMLToJSON.java [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+            private Seq<Map<QName,XSParticle>> expected = List.empty();
+            private Option<QName> continueArray = none();
+            private boolean expectAttributes = false;
+            
+            {
+                setHandler(in, new AbstractInHandler() {
+
+                    @Override
+                    public void onPush() throws Exception {
+                        XMLEvent evt = grab(in);
```
Comment (alar17) [inline ts-reaktive-marshal-xerces/src/main/java/com/tradeshift/reaktive/marshal/convert/XMLToJSON.java]: Reading this block of code (onPush) with lots of "if-else"s was not easy to read. I lost several times in the middle of it. Maybe adding a single short comment at the beginning of each part like root, start element, end element, array and etc can improve the readability of the code.

File: ts-reaktive-marshal-xerces/src/test/java/com/tradeshift/reaktive/marshal/convert/XMLToJSONSpec.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+import com.tradeshift.reaktive.marshal.stream.AaltoReader;
+import com.tradeshift.reaktive.marshal.stream.JacksonWriter;
+import com.tradeshift.reaktive.testkit.SharedActorSystemSpec;
+
+import akka.stream.javadsl.StreamConverters;
+import akka.util.ByteString;
+
+@RunWith(CuppaRunner.class)
+public class XMLToJSONSpec extends SharedActorSystemSpec {
```
Comment (jypma) [inline ts-reaktive-marshal-xerces/src/test/java/com/tradeshift/reaktive/marshal/convert/XMLToJSONSpec.java]: agreed, adding test.

File: ts-reaktive-marshal-xerces/src/main/java/com/tradeshift/reaktive/marshal/convert/XMLToJSON.java [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+    }
+    
+    private boolean isMultiValued(XSParticle particle) {
+        return particle.getMaxOccursUnbounded() || particle.getMaxOccurs() > 1;
+    }
+    
+    private boolean hasAttributes(XSElementDeclaration elmt) {
+        if (elmt.getTypeDefinition() instanceof XSComplexTypeDefinition) {
+            //Yes, we don't need to wrap this in an object if no attributes can ever occur. But Jackson does, so we must do so too.
+            //XSComplexTypeDefinition def = (XSComplexTypeDefinition) elmt.getTypeDefinition();
```
Comment (jypma) [inline ts-reaktive-marshal-xerces/src/main/java/com/tradeshift/reaktive/marshal/convert/XMLToJSON.java]: it was supposed to indicate a nicer, future, format we may implement. I'll update the comment to reflect that.

**Comment by jypma** [comment]
@alar17 updated according to your comments. Please check again.

**Pull request merged**